### PR TITLE
Remove work_package_sharing feature flag

### PR DESCRIPTION
**Pull request by klaustopher**
Removes the feature flag that guarded the sharing feature

---

Implements https://community.openproject.org/projects/openproject/work_packages/50260

### CODE REVIEW

File: frontend/src/app/features/in-app-notifications/center/menu/menu.component.ts [hunk 1]
```diff
@@ -172,9 +172,7 @@ export class IanMenuComponent implements OnInit {
     readonly bannersService:BannersService,
     readonly configurationService:ConfigurationService,
   ) {
-    if (this.configurationService.activeFeatureFlags.includes('workPackageSharing')) {
       this.reasonMenuItems.push(this.sharedMenuItem);
```
Comment (ulferts) [inline frontend/src/app/features/in-app-notifications/center/menu/menu.component.ts]: The contents of `this.sharedMenuItem` can now be statically moved to the `reasonMenuItems` array.

File: frontend/src/app/features/work-packages/components/wp-details/wp-details-toolbar.component.ts [hunk 2]
```diff
@@ -52,6 +52,6 @@ export class WorkPackageSplitViewToolbarComponent {
     readonly halEditing:HalResourceEditingService,
     readonly configurationService:ConfigurationService,
   ) {
-    this.displayShareButton = this.configurationService.activeFeatureFlags.includes('workPackageSharing');
+    this.displayShareButton = true;
```
Comment (ulferts) [inline frontend/src/app/features/work-packages/components/wp-details/wp-details-toolbar.component.ts]: The `displayShareButton` variable can be removed altogether as it has now become static. This requires removing the `*ngIf="displayShareButton" conditions from the templates.

**Review by ulferts — APPROVED** [review]
Just tiny things that can be improved, @klaustopher. Feel free to merge on your own once they are addressed.

File: frontend/src/app/features/work-packages/components/wp-details/wp-details-toolbar.component.ts (hunk 2 above)
Comment (klaustopher) [reply frontend/src/app/features/work-packages/components/wp-details/wp-details-toolbar.component.ts]: There is still one instance where the button is dependent on a capability, can we really remove it everywhere?

File: frontend/src/app/features/in-app-notifications/center/menu/menu.component.ts (hunk 1 above)
Comment (klaustopher) [reply frontend/src/app/features/in-app-notifications/center/menu/menu.component.ts]: done

File: frontend/src/app/features/work-packages/components/wp-details/wp-details-toolbar.component.ts (hunk 2 above)
Comment (klaustopher) [reply frontend/src/app/features/work-packages/components/wp-details/wp-details-toolbar.component.ts]: Actually I have removed the setting from all other places. In this place it should actually be based on the capabilities for the current work package. But I am not sure if we cann add this here. @aaron-contreras As you added the capabilities check to the other place, could you take a look if we can add this here as well?

Comment (oliverguenther) [reply frontend/src/app/features/work-packages/components/wp-details/wp-details-toolbar.component.ts]: Done https://github.com/opf/openproject/pull/14393/commits/1379e5833bd29be812042cc82df587ef72f58ef2

**Review by oliverguenther — APPROVED** [review]

File: frontend/src/app/features/work-packages/routing/wp-full-view/wp-full-view.component.ts [hunk 3]
```diff
@@ -107,8 +107,7 @@ export class WorkPackagesFullViewComponent extends WorkPackageSingleViewBase imp
     this.isWatched = Object.prototype.hasOwnProperty.call(wp, 'unwatch');
     this.displayWatchButton = Object.prototype.hasOwnProperty.call(wp, 'unwatch') || Object.prototype.hasOwnProperty.call(wp, 'watch');
     this.displayTimerButton = Object.prototype.hasOwnProperty.call(wp, 'logTime');
-    this.displayShareButton$ = this.configurationService.activeFeatureFlags.includes('workPackageSharing')
-      && this.currentUserService.hasCapabilities$('work_package_shares/index', wp.project.id);
+    this.displayShareButton$ = this.currentUserService.hasCapabilities$('work_package_shares/index', wp.project.id);
```
Comment (aaron-contreras) [inline frontend/src/app/features/work-packages/routing/wp-full-view/wp-full-view.component.ts:110]: 👍

File: spec/features/work_packages/share_spec.rb [hunk 4]
```diff
@@ -316,16 +314,6 @@ def inherited_member_roles(group:)
...
-  context 'without the feature flag enabled', with_flag: { work_package_sharing: false } do
-    let(:sharer_role) do
-      create(:project_role,
-             permissions: %i(view_work_packages
-                             view_shared_work_packages))
-    end
-
-    it_behaves_like "'Share' button is not rendered"
-  end
-
```
Comment (aaron-contreras) [inline spec/features/work_packages/share_spec.rb:328]: 👍

**Review by aaron-contreras — APPROVED** [review]
Good to go!

**Pull request merged**